examples/i2c_oled.py

```diff
@@ -0,0 +1,237 @@
+font = ImageFont.truetype("/usr/share/fonts/dejavu/DejaVuSans.ttf", 12)
+import os, os.path
+
+"""
+This example sets up and control an Oled screen.
+"""
+
+
+class Oled(_SSD1306):
+    def __init__(
+        self,
+        width,
+        height,
+        board,
+        addr=0x3C,
+        external_vcc=False,
+        reset=None,
+    ):
+        self.board = board
+        self.init_awaits = []
+        self.write_commands = []
+        self.addr = addr
+        self.temp = bytearray(2)
+        self.i2c_port = 0
+        self.failed = False
+        # Add an extra byte to the data buffer to hold an I2C data/command byte
+        # to use hardware-compatible I2C transactions.  A memoryview of the
+        # buffer is used to mask this byte from the framebuffer operations
+        # (without a major memory hit as memoryview doesn't copy to a separate
+        # buffer).
+        self.buffer = bytearray(((height // 8) * width) + 1)
+        # self.buffer = bytearray(16)
+        # self.buffer[0] = 0x40  # Set first byte of data buffer to Co=0, D/C=1
+        pin_numbers = {"sda": 4, "scl": 5}
+        self.init_awaits.append(self.board.set_pin_mode_i2c(
+            i2c_port=0,
+            sda_gpio=pin_numbers["sda"],
+            scl_gpio=pin_numbers["scl"],
+        ))
+        time.sleep(1)
+        super().__init__(
+            memoryview(self.buffer)[1:],
+            width,
+            height,
+            external_vcc=external_vcc,
+            reset=reset,
+            page_addressing=False,
+        )
+
+    async def start(self):
+        # Adafruit SSD1306 initializes the oled in the constructor, where async is not allowed.
+        # To fix this, add all commands to a list and process them afterwards.
+        for ev in self.init_awaits:
+            await ev
+        for cmd in self.write_commands:
+            out = await self.board.i2c_write(60, cmd, i2c_port=self.i2c_port)
+            if(not out):
+                print("write failed start")
+                self.failed=True
+                return
+
+    # TODO: make faster with asyncio
+    async def set_oled_image_service(self, type, value):
+        if(self.failed):
+            print("oled writing failed")
+            return
+        if type == "text":
+            text = value.replace("\\n", "\n")
+            image = Image.new("1", (128, 64))
+            draw = ImageDraw.Draw(image)
+            split_text = text.splitlines()
+            lines = []
+            for i in split_text:
+                lines.extend(textwrap.wrap(i, width=20))
+
+            y_text = 1
+            for line in lines:
+                width, height = font.getsize(line)
+                draw.text((1, y_text), line, font=font, fill=255)
+                y_text += height
+            self.image(image)
+            await self.show_async()
+
+        if type == "image":
+            self.show_png(
+                "/usr/local/src/mirte/mirte-oled-images/images/" + value + ".png"
```

Comment:
This folder will not be found on non  Mirte setups

Reply:
Good catch, removed the prefix